File: Leetcode-one-by-one/src/Sort Integers by The Number of 1 Bits.swift
class Solution {
    func sortByBits(_ arr: [Int]) -> [Int] {
        return arr.sorted { a, b in
            let bitsA = a.nonzeroBitCount
            let bitsB = b.nonzeroBitCount
            
            if bitsA != bitsB {
                return bitsA < bitsB
            }
            return a < b
        }
    }
}

File: Leetcode-one-by-one/src/Sort Integers by The Number of 1 Bits.java
import java.util.*;

class Solution {
    public int[] sortByBits(int[] arr) {
        Integer[] nums = Arrays.stream(arr).boxed().toArray(Integer[]::new);

        Arrays.sort(nums, (a, b) -> {
            int bitsA = Integer.bitCount(a);
            int bitsB = Integer.bitCount(b);
            if (bitsA != bitsB) {
                return bitsA - bitsB;
            }
            return a - b;
        });

        return Arrays.stream(nums).mapToInt(Integer::intValue).toArray();
    }
}
